jhbuild/modtypes/__init__.py

```diff
@@ -300,6 +302,47 @@ def _process_install_files(self, installroot, curdir, prefix, errors):
                 errors.append(str(e))
         return num_copied
 
+    def _do_strip(self, destdir_prefix, contents):
+        # first find exec
+        files_to_strip = []
+
+        # HACK from orininal bash script.
+        excluded_packages = ['scipy', 'PIL']
+        excluded_libs = ['libQt5']
+        excluded_file_info = ['ELF', ', not stripped']
+
+        # filter out files to strip
+        for filename in contents:
+            if os.access(filename, os.X_OK) or 'so' in filename.split(os.path.extsep):
+                splited_filename = filename.split(os.path.sep)
+                if any([s == e for s in splited_filename for e in excluded_packages]):
+                    continue
+                if any([e in filename for e in excluded_libs]):
+                    continue
+                file_info = subprocess.check_output(['file', '-b', filename])
+                if any([e in file_info for e in excluded_file_info]):
+                    continue
+                files_to_strip.append(filename)
+
+        # do strip
+        for filename in files_to_strip:
+            dirname = os.path.dirname(filename)
+            debug_dir = os.path.join(destdir_prefix, 'debug')
+            dir_in_debug = os.path.join(debug_dir, dirname)
+            if not os.path.exists(dir_in_debug):
+                os.makedirs(dir_in_debug)
+            dotdebug_link =os.path.join(destdir_prefix, dirname, '.debug')
+            if os.path.exists(dotdebug_link):
+                os.remove(dotdebug_link)
+            os.symlink(os.path.join(debug_dir, dirname), dotdebug_link)
```

Comment:
So, because now multiple jhbuild modules will produce the same file, the link `.debug` in a folder. I would like you to do it in a different approach, instead of creating this folder `.debug` link.

Please create the `filename.debug` link and link it to `/opt/debug/directory/filename.debug`, so each package will produce the `xxx.so` plus the `xxx.so.debug` link file, and not directory `.debug` link.

See https://sourceware.org/gdb/onlinedocs/gdb/Separate-Debug-Files.html